src/weights.jl

```diff
@@ -281,6 +288,9 @@ Base.mean{T<:Number,W<:Real}(A::AbstractArray{T}, w::WeightVec{W}, dim::Int) =
 
 
 ###### Weighted median #####
+function Base.median(v::AbstractArray, w::WeightVec)
```

Comment:
I needed to added this because [this](https://github.com/JuliaStats/StatsBase.jl/blob/master/test/weights.jl#L239) test was failing with a `BoundsError` instead as it was able to dispatch to a method in base.

```julia
julia> median([4 3 2 1 0], weights(wt))
ERROR: BoundsError: attempt to access 2-element Array{Any,1} at index [3]
 in mapslices(::Base.#median!, ::Array{Int64,2}, ::StatsBase.WeightVec{Int64,Array{Int64,1},Int64}) at ./abstractarray.jl:1619
 in median(::Array{Int64,2}, ::StatsBase.WeightVec{Int64,Array{Int64,1},Int64}) at ./statistics.jl:579```

I figured it was better to manually throw a `MethodError` rather than letting it hit the `BoundsError` in that case.